cpp/include/cuspatial/experimental/detail/geometry_collection/multipoint_ref.cuh

```diff
@@ -37,4 +39,17 @@ CUSPATIAL_HOST_DEVICE auto multipoint_ref<VecIterator>::point_end() const
   return _points_end;
 }
 
+template <typename VecIterator>
+CUSPATIAL_HOST_DEVICE auto multipoint_ref<VecIterator>::num_points() const
+{
+  return thrust::distance(_points_begin, _points_end);
+}
+
+template <typename VecIterator>
+template <typename IndexType>
+CUSPATIAL_HOST_DEVICE auto multipoint_ref<VecIterator>::operator[](IndexType i)
```

Comment:
Just thinking about host / device here. Is this really intended to work on the host? What if it is called from host when the iterator points to device memory?

Reply:
This can work on host, if all vectors are host vectors. Although I don't really see a use case at the moment. Up till now I mostly rely on developer's prudence to not access device memory from host.